### ISSUE TITLE
macOS 10.13 with newest ct-ng reports "unsupported option '-print-multi-os-directory' "

### ISSUE THREAD

**Issue by huming2207** [crosstool-ng/crosstool-ng#844]
Hello there,

I've installed the newest crosstool-ng from your github repo, and when it starts to build gcc, it reports an error which looks like this:

```
[INFO ]  =================================================================
[INFO ]  Installing final gcc compiler
[EXTRA]    Configuring final gcc compiler
[EXTRA]    Building final gcc compiler
[ERROR]    clang: error: unsupported option '-print-multi-os-directory'
[ERROR]    clang: error: no input files
[ERROR]    /Volumes/Mokee/ct-ng/.build/aarch64-hu-linux-android/src/gcc/libstdc++-v3/libsupc++/exception:143:10: fatal error: bits/nested_exception.h: No such file or directory
[ERROR]    make[5]: *** [aarch64-hu-linux-android/bits/stdc++.h.gch/O2ggnu++0x.gch] Error 1
[ERROR]    make[5]: *** Waiting for unfinished jobs....
[ERROR]    make[4]: *** [all-recursive] Error 1
[ERROR]    make[3]: *** [all] Error 2
[ERROR]    make[2]: *** [all-target-libstdc++-v3] Error 2
[ERROR]    make[1]: *** [all] Error 2
[ERROR]
[ERROR]  >>
[ERROR]  >>  Build failed in step 'Installing final gcc compiler'
[ERROR]  >>        called in step '(top-level)'
[ERROR]  >>
[ERROR]  >>  Error happened in: CT_DoExecLog[scripts/functions@372]
[ERROR]  >>        called from: do_gcc_backend[scripts/build/cc/gcc.sh@1140]
[ERROR]  >>        called from: do_cc_for_host[scripts/build/cc/gcc.sh@824]
[ERROR]  >>        called from: main[scripts/crosstool-NG.sh@675]
[ERROR]  >>
[ERROR]  >>  For more info on this error, look at the file: 'build.log'
[ERROR]  >>  There is a list of known issues, some with workarounds, in:
[ERROR]  >>      https://crosstool-ng.github.io/docs/known-issues/
[ERROR]  >>
[ERROR]  >> NOTE: Your configuration includes features marked EXPERIMENTAL.
[ERROR]  >> Before submitting a bug report, try to reproduce it without enabling
[ERROR]  >> any experimental features. Otherwise, you'll need to debug it
[ERROR]  >> and present an explanation why it is a bug in crosstool-NG - or
[ERROR]  >> preferably, a fix.
[ERROR]  >>
[ERROR]  >>  If you feel this is a bug in crosstool-NG, report it at:
[ERROR]  >>      https://github.com/crosstool-ng/crosstool-ng/issues/
[ERROR]  >>
[ERROR]  >>  Make sure your report includes all the information pertinent to this issue.
[ERROR]  >>  Read the bug reporting guidelines here:
[ERROR]  >>      http://crosstool-ng.github.io/support/
[ERROR]
[ERROR]  (elapsed: 8:32.69)
[08:33] / make: *** [build] Error 2
```

**It looks like `clang` does not support this argument `-print-multi-os-directory`. By the way I would like to know that can I use the gcc (not clang) from Homebrew? How do I specify the host toolchain?**

Regards,
Jackson

**Comment by stilor** [comment]
-print-multi-os-directory warning comes from libiberty configure and is harmless. Read http://crosstool-ng.github.io/docs/os-setup/

Both clang and gcc should be usable as the build compiler.

The real error is the missing nested_exception.h header. However, you didn't attach your .config so I cannot even try to reproduce this issue. I don't have time to debug/test on macos right now, though, so if it is not reproducible on ubuntu, you'd have to wait a couple of weeks (or debug it yourself).

**Comment by huming2207** [comment]
Ah sorry I forgot to paste the .config file. Sorry about that.
I'll try debug by myself later because currently I don't have any mac usable now.

**Comment by huming2207** [comment]
@stilor Sorry dude I've had some attempts a moment ago. 

Looks like something went wrong with Linaro GCC 7.1. If I compile the stock GNU GCC 7.2 it works fine, no issue at all...

![image](https://user-images.githubusercontent.com/4463497/31584118-bbe769a0-b1f3-11e7-9718-d6d450ba5c82.png)

Here is my build log (the successful one only currently) and build configuration. The only difference is the GCC version.

Btw, please rename this log file to `build.tar.xz` as Github limits the file extensions now.

[build.tar.xz.zip](https://github.com/crosstool-ng/crosstool-ng/files/1385363/build.tar.xz.zip)

**Comment by huming2207** [comment]
Ah wait not "dude"... I've edited... So sorry about that... I was messaging to someone else and forgot something here...

**Comment by tofutim** [comment]
I got 
```
clang: error: unsupported option '-print-multi-os-directory'
```
today attempting armv6-rpi-linux on OSX 10.13

**Comment by stilor** [comment]
Do you have a linaro version of gcc, same as the original reporter?

**Comment by tofutim** [comment]
i think the problem is actually gmake, which is remake on my end
maybe that error is ok

**Comment by avisiedo** [comment]
Same issue here. Building crosstool for raspberry pi 3 into macos High Sierra 10.13.6

```
[INFO ]  =================================================================
[INFO ]  Installing final gcc compiler
[EXTRA]    Configuring final gcc compiler
[EXTRA]    Building final gcc compiler
[ERROR]    clang: error: unsupported option '-print-multi-os-directory'
[ERROR]    clang: error: no input files
[ERROR]    /Users/avisiedo/aarch64-rpi3-linux-gnueabi/.build/aarch64-rpi3-linux-gnueabi/build/build-cc-gcc-final/aarch64-rpi3-linux-gnueabi/libstdc++-v3/include/ext/alloc_traits.h:36:32: fatal error: bits/alloc_traits.h: No such file or directory
[ERROR]    make[5]: *** [Makefile:1719: aarch64-rpi3-linux-gnueabi/bits/stdc++.h.gch/O2ggnu++0x.gch] Error 1
[ERROR]    make[5]: *** Waiting for unfinished jobs....
[ERROR]    make[4]: *** [Makefile:507: all-recursive] Error 1
[ERROR]    make[3]: *** [Makefile:414: all] Error 2
[ERROR]    make[2]: *** [Makefile:10313: all-target-libstdc++-v3] Error 2
[ERROR]    make[1]: *** [Makefile:909: all] Error 2
```

This is the error registered into build.log:

```
[ALL  ]    /Users/avisiedo/aarch64-rpi3-linux-gnueabi/.build/aarch64-rpi3-linux-gnueabi/build/build-cc-gcc-final/./gcc/xgcc -shared-libgcc -B/Users/avisiedo/aarch64-rpi3-linux-gnueabi/.build/aarch64-rpi3-linux-gnueabi/build/build-cc-gcc-f
inal/./gcc -nostdinc++ -L/Users/avisiedo/aarch64-rpi3-linux-gnueabi/.build/aarch64-rpi3-linux-gnueabi/build/build-cc-gcc-final/aarch64-rpi3-linux-gnueabi/libstdc++-v3/src -L/Users/avisiedo/aarch64-rpi3-linux-gnueabi/.build/aarch64-rpi3-li
nux-gnueabi/build/build-cc-gcc-final/aarch64-rpi3-linux-gnueabi/libstdc++-v3/src/.libs -L/Users/avisiedo/aarch64-rpi3-linux-gnueabi/.build/aarch64-rpi3-linux-gnueabi/build/build-cc-gcc-final/aarch64-rpi3-linux-gnueabi/libstdc++-v3/libsupc
++/.libs -B/Users/avisiedo/aarch64-rpi3-linux-gnueabi/crosstools/aarch64-rpi3-linux-gnueabi/aarch64-rpi3-linux-gnueabi/bin/ -B/Users/avisiedo/aarch64-rpi3-linux-gnueabi/crosstools/aarch64-rpi3-linux-gnueabi/aarch64-rpi3-linux-gnueabi/lib/
 -isystem /Users/avisiedo/aarch64-rpi3-linux-gnueabi/crosstools/aarch64-rpi3-linux-gnueabi/aarch64-rpi3-linux-gnueabi/include -isystem /Users/avisiedo/aarch64-rpi3-linux-gnueabi/crosstools/aarch64-rpi3-linux-gnueabi/aarch64-rpi3-linux-gnueabi/sys-include    -x c++-header -nostdinc++ -mlittle-endian   -mcpu=cortex-a53     -D_GNU_SOURCE -g -Os  -I/Users/avisiedo/aarch64-rpi3-linux-gnueabi/.build/aarch64-rpi3-linux-gnueabi/build/build-cc-gcc-final/aarch64-rpi3-linux-gnueabi/libstdc++-v3/include/aarch64-rpi3-linux-gnueabi -I/Users/avisiedo/aarch64-rpi3-linux-gnueabi/.build/aarch64-rpi3-linux-gnueabi/build/build-cc-gcc-final/aarch64-rpi3-linux-gnueabi/libstdc++-v3/include -I/Users/avisiedo/aarch64-rpi3-linux-gnueabi/.build/src/gcc-6.3.0/libstdc++-v3/libsupc++  -O2 -g /Users/avisiedo/aarch64-rpi3-linux-gnueabi/.build/src/gcc-6.3.0/libstdc++-v3/include/precompiled/stdc++.h -o aarch64-rpi3-linux-gnueabi/bits/stdc++.h.gch/O2g.gch
[ALL  ]    In file included from /Users/avisiedo/aarch64-rpi3-linux-gnueabi/.build/aarch64-rpi3-linux-gnueabi/build/build-cc-gcc-final/aarch64-rpi3-linux-gnueabi/libstdc++-v3/include/bits/basic_string.h:40:0,
[ALL  ]                     from /Users/avisiedo/aarch64-rpi3-linux-gnueabi/.build/aarch64-rpi3-linux-gnueabi/build/build-cc-gcc-final/aarch64-rpi3-linux-gnueabi/libstdc++-v3/include/string:52,
[ALL  ]                     from /Users/avisiedo/aarch64-rpi3-linux-gnueabi/.build/aarch64-rpi3-linux-gnueabi/build/build-cc-gcc-final/aarch64-rpi3-linux-gnueabi/libstdc++-v3/include/bits/locale_classes.h:40,
[ALL  ]                     from /Users/avisiedo/aarch64-rpi3-linux-gnueabi/.build/aarch64-rpi3-linux-gnueabi/build/build-cc-gcc-final/aarch64-rpi3-linux-gnueabi/libstdc++-v3/include/bits/ios_base.h:41,
[ALL  ]                     from /Users/avisiedo/aarch64-rpi3-linux-gnueabi/.build/aarch64-rpi3-linux-gnueabi/build/build-cc-gcc-final/aarch64-rpi3-linux-gnueabi/libstdc++-v3/include/ios:42,
[ALL  ]                     from /Users/avisiedo/aarch64-rpi3-linux-gnueabi/.build/aarch64-rpi3-linux-gnueabi/build/build-cc-gcc-final/aarch64-rpi3-linux-gnueabi/libstdc++-v3/include/istream:38,
[ALL  ]                     from /Users/avisiedo/aarch64-rpi3-linux-gnueabi/.build/aarch64-rpi3-linux-gnueabi/build/build-cc-gcc-final/aarch64-rpi3-linux-gnueabi/libstdc++-v3/include/sstream:38,
[ALL  ]                     from /Users/avisiedo/aarch64-rpi3-linux-gnueabi/.build/aarch64-rpi3-linux-gnueabi/build/build-cc-gcc-final/aarch64-rpi3-linux-gnueabi/libstdc++-v3/include/complex:45,
[ALL  ]                     from /Users/avisiedo/aarch64-rpi3-linux-gnueabi/.build/aarch64-rpi3-linux-gnueabi/build/build-cc-gcc-final/aarch64-rpi3-linux-gnueabi/libstdc++-v3/include/ccomplex:39,
[ALL  ]                     from /Users/avisiedo/aarch64-rpi3-linux-gnueabi/.build/src/gcc-6.3.0/libstdc++-v3/include/precompiled/stdc++.h:52:
[ERROR]    /Users/avisiedo/aarch64-rpi3-linux-gnueabi/.build/aarch64-rpi3-linux-gnueabi/build/build-cc-gcc-final/aarch64-rpi3-linux-gnueabi/libstdc++-v3/include/ext/alloc_traits.h:36:32: fatal error: bits/alloc_traits.h: No such file or directory
[ALL  ]     # include <bits/alloc_traits.h>
[ALL  ]                                    ^
[ALL  ]    compilation terminated.
[ERROR]    make[5]: *** [Makefile:1719: aarch64-rpi3-linux-gnueabi/bits/stdc++.h.gch/O2ggnu++0x.gch] Error 1
[ERROR]    make[5]: *** Waiting for unfinished jobs....
[ALL  ]    make[5]: Leaving directory '/Users/avisiedo/aarch64-rpi3-linux-gnueabi/.build/aarch64-rpi3-linux-gnueabi/build/build-cc-gcc-final/aarch64-rpi3-linux-gnueabi/libstdc++-v3/include'
[ERROR]    make[4]: *** [Makefile:507: all-recursive] Error 1
[ALL  ]    make[4]: Leaving directory '/Users/avisiedo/aarch64-rpi3-linux-gnueabi/.build/aarch64-rpi3-linux-gnueabi/build/build-cc-gcc-final/aarch64-rpi3-linux-gnueabi/libstdc++-v3'
[ERROR]    make[3]: *** [Makefile:414: all] Error 2
[ALL  ]    make[3]: Leaving directory '/Users/avisiedo/aarch64-rpi3-linux-gnueabi/.build/aarch64-rpi3-linux-gnueabi/build/build-cc-gcc-final/aarch64-rpi3-linux-gnueabi/libstdc++-v3'
[ERROR]    make[2]: *** [Makefile:10313: all-target-libstdc++-v3] Error 2
[ALL  ]    make[2]: Leaving directory '/Users/avisiedo/aarch64-rpi3-linux-gnueabi/.build/aarch64-rpi3-linux-gnueabi/build/build-cc-gcc-final'
[ERROR]    make[1]: *** [Makefile:909: all] Error 2
[ALL  ]    make[1]: Leaving directory '/Users/avisiedo/aarch64-rpi3-linux-gnueabi/.build/aarch64-rpi3-linux-gnueabi/build/build-cc-gcc-final'
[ERROR]  
```

Launching the command that failed by hand in this way, worked properly:

```
Isaac-Newton:aarch64-rpi3-linux-gnueabi avisiedo$ find . -name '*.gch'
./.build/aarch64-rpi3-linux-gnueabi/build/build-cc-gcc-final/aarch64-rpi3-linux-gnueabi/libstdc++-v3/include/aarch64-rpi3-linux-gnueabi/bits/stdc++.h.gch
./.build/aarch64-rpi3-linux-gnueabi/build/build-cc-gcc-final/aarch64-rpi3-linux-gnueabi/libstdc++-v3/include/aarch64-rpi3-linux-gnueabi/bits/stdc++.h.gch/O2g.gch
Isaac-Newton:aarch64-rpi3-linux-gnueabi avisiedo$ pushd .build/aarch64-rpi3-linux-gnueabi/build/build-cc-gcc-final/aarch64-rpi3-linux-gnueabi/libstdc++-v3/include/
~/aarch64-rpi3-linux-gnueabi/.build/aarch64-rpi3-linux-gnueabi/build/build-cc-gcc-final/aarch64-rpi3-linux-gnueabi/libstdc++-v3/include ~/aarch64-rpi3-linux-gnueabi
Isaac-Newton:include avisiedo$ /Users/avisiedo/aarch64-rpi3-linux-gnueabi/.build/aarch64-rpi3-linux-gnueabi/build/build-cc-gcc-final/./gcc/xgcc -shared-libgcc -B/Users/avisiedo/aarch64-rpi3-linux-gnueabi/.build/aarch64-rpi3-linux-gnueabi/build/build-cc-gcc-final/./gcc -nostdinc++ -L/Users/avisiedo/aarch64-rpi3-linux-gnueabi/.build/aarch64-rpi3-linux-gnueabi/build/build-cc-gcc-final/aarch64-rpi3-linux-gnueabi/libstdc++-v3/src -L/Users/avisiedo/aarch64-rpi3-linux-gnueabi/.build/aarch64-rpi3-linux-gnueabi/build/build-cc-gcc-final/aarch64-rpi3-linux-gnueabi/libstdc++-v3/src/.libs -L/Users/avisiedo/aarch64-rpi3-linux-gnueabi/.build/aarch64-rpi3-linux-gnueabi/build/build-cc-gcc-final/aarch64-rpi3-linux-gnueabi/libstdc++-v3/libsupc++/.libs -B/Users/avisiedo/aarch64-rpi3-linux-gnueabi/crosstools/aarch64-rpi3-linux-gnueabi/aarch64-rpi3-linux-gnueabi/bin/ -B/Users/avisiedo/aarch64-rpi3-linux-gnueabi/crosstools/aarch64-rpi3-linux-gnueabi/aarch64-rpi3-linux-gnueabi/lib/ -isystem /Users/avisiedo/aarch64-rpi3-linux-gnueabi/crosstools/aarch64-rpi3-linux-gnueabi/aarch64-rpi3-linux-gnueabi/include -isystem /Users/avisiedo/aarch64-rpi3-linux-gnueabi/crosstools/aarch64-rpi3-linux-gnueabi/aarch64-rpi3-linux-gnueabi/sys-include    -x c++-header -nostdinc++ -mlittle-endian   -mcpu=cortex-a53     -D_GNU_SOURCE -g -Os  -I/Users/avisiedo/aarch64-rpi3-linux-gnueabi/.build/aarch64-rpi3-linux-gnueabi/build/build-cc-gcc-final/aarch64-rpi3-linux-gnueabi/libstdc++-v3/include/aarch64-rpi3-linux-gnueabi -I/Users/avisiedo/aarch64-rpi3-linux-gnueabi/.build/aarch64-rpi3-linux-gnueabi/build/build-cc-gcc-final/aarch64-rpi3-linux-gnueabi/libstdc++-v3/include -I/Users/avisiedo/aarch64-rpi3-linux-gnueabi/.build/src/gcc-6.3.0/libstdc++-v3/libsupc++  -O2 -g /Users/avisiedo/aarch64-rpi3-linux-gnueabi/.build/src/gcc-6.3.0/libstdc++-v3/include/precompiled/stdc++.h -o aarch64-rpi3-linux-gnueabi/bits/stdc++.h.gch/O2g.gch
Isaac-Newton:include avisiedo$ popd
~/aarch64-rpi3-linux-gnueabi
```

It seems that some command did not build properly the output path, or the include list is not completed.

**Comment by stilor** [comment]
Looks like the APFS issue described in our docs: http://crosstool-ng.github.io/docs/os-setup/

All, please read the docs before posting the bugs.

**Comment by roolebo** [comment]
It seems to be fixed (workarounded) in GCC 6.5, 7.4 and 8.1. There's an option to use 8.2 in master branch of crosstool-ng. Nonetheless, homebrew version of crosstool-ng is more than 1,5 years old and does include only GCC 6.3.0.
